etcdserver/api/v3rpc/watch.go

```diff
@@ -180,6 +200,19 @@ func (sws *serverWatchStream) recvLoop() error {
 				// support  >= key queries
 				creq.RangeEnd = []byte{}
 			}
+
+			if !sws.isWatchPermitted(creq) {
+				wr := &pb.WatchResponse{
```

Comment:
This should fill out all of the fields like the regular error path below. Older clients will be confused by this behavior since they'll get id=0 and not know about permission_denied and think the watch was successfully created.